lib/sentry.dart

```diff
@@ -494,3 +503,445 @@ class User {
     };
   }
 }
+
+/// The context interfaces provide additional context data.
+/// Typically this is data related to the current user,
+/// the current HTTP request.
+class Contexts {
+  /// This describes the device that caused the event.
+  final Device device;
+
+  /// Describes the operating system on which the event was created.
+  /// In web contexts, this is the operating system of the browse
+  /// (normally pulled from the User-Agent string).
+  final Os os;
+
+  /// Describes a runtime in more detail.
+  /// Typically this context is used multiple times if multiple runtimes
+  /// are involved (for instance if you have a JavaScript application running
+  /// on top of JVM).
+  final List<Runtime> runtimes;
+
+  /// Describes a runtime in more detail.
+  /// Typically this context is used multiple times if multiple runtimes
+  /// are involved (for instance if you have a JavaScript application running
+  /// on top of JVM).
+  final App app;
+
+  /// Carries information about the browser or user agent for web-related
+  /// errors. This can either be the browser this event ocurred in, or the user
+  /// agent of a web request that triggered the event.
+  final Browser browser;
+
+  const Contexts({this.device, this.os, this.runtimes, this.app, this.browser});
+
+  /// Produces a [Map] that can be serialized to JSON.
+  Map<String, dynamic> toJson() {
+    final Map<String, dynamic> json = {};
+
+    Map<String, dynamic> deviceMap;
+    if (device != null && (deviceMap = device.toJson()).isNotEmpty) {
+      json['device'] = deviceMap;
+    }
+
+    Map<String, dynamic> osMap;
+    if (os != null && (osMap = os.toJson()).isNotEmpty) {
+      json['os'] = osMap;
+    }
+
+    Map<String, dynamic> appMap;
+    if (app != null && (appMap = app.toJson()).isNotEmpty) {
+      json['app'] = appMap;
+    }
+
+    Map<String, dynamic> browserMap;
+    if (browser != null && (browserMap = browser.toJson()).isNotEmpty) {
+      json['browser'] = browserMap;
+    }
+
+    if (runtimes != null) {
+      if (runtimes.length == 1) {
+        final runtime = runtimes[0];
```

Comment:
This package follows the [Flutter Style Guide](https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo), where we [always specify types](https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#avoid-using-var), even for `final`s.

(this comment applies throughout the PR)